storage/shard.go

```diff
@@ -532,96 +573,107 @@ func (s *dbShard) tickAndExpire(
 	tickSleepBatch := s.currRuntimeOptions.tickSleepSeriesBatchSize
 	tickSleepPerSeries := s.currRuntimeOptions.tickSleepPerSeries
 	s.RUnlock()
-	s.forEachShardEntry(func(entry *dbShardEntry) bool {
-		if i > 0 && i%tickSleepBatch == 0 {
-			// NB(xichen): if the tick is cancelled, we bail out immediately.
-			// The cancellation check is performed on every batch of entries
-			// instead of every entry to reduce load.
-			if c.IsCancelled() {
-				return false
+	s.forEachShardEntryBatch(func(currEntries []*dbShardEntry) bool {
+		var expired []*dbShardEntry
```

Comment:
`expired = expired[:0]` and reuse this slice perhaps?

Reply:
sure thing.

Reply:
done